### PR TITLE
feat: Add configurable `artifactErrorWarning` user string

### PR DESCRIPTION
**Pull request by hesstobi**
## Changes

This PR introduces a new configurable user string `artifactErrorWarning` that allows users to customize the warning message displayed in PR comments when artifact errors occur during dependency updates.

**Key changes:**
- Added `artifactErrorWarning` to the `userStrings` configuration object with default value: `"You probably do not want to merge this PR as-is."`
- Modified artifact error comment generation in `lib/workers/repository/update/branch/index.ts` to use the configurable user string with template compilation support
- Updated documentation to include the new user string option with examples
- Added comprehensive test coverage for default, custom, and templated messages

## Context

This change was implemented to support GitLab's quick action functionality, specifically to enable adding labels using the `/label ~my-nice-label` command in PR comments. By making the artifact error warning message configurable, users can now include GitLab quick actions directly in their custom warning messages, allowing for automated label management when artifact errors occur.

**Example use case:**
```json
{
  "userStrings": {
    "artifactErrorWarning": "You probably do not want to merge this PR as-is.\n\n/label ~artifact-error ~needs-review"
  }
}
```

This enables automatic labeling of PRs with artifact errors for improved workflow management in GitLab environments.

## Documentation (please check one with an [x])

- [x] I have updated the documentation, or
- [ ] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [ ] Code inspection only, or
- [x] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [x] Both unit tests + ran on a real repository

**Test coverage includes:**
- Verification of default message behavior (backward compatibility)
- Testing custom user-defined messages
- Testing template variable substitution (e.g., `{{manager}}`)

### CODE REVIEW

**Comment by hesstobi** [comment]
Hi team!

I've implemented this feature to enable GitLab quick actions (like `/label ~my-nice-label`) in artifact error comments, which should improve workflow automation for GitLab users.

I'd appreciate any feedback on the implementation:
- Is the approach with `userStrings` the right way to go for this type of customization?
- Are there any edge cases I should consider or additional test scenarios needed?
- Should I add any additional documentation or examples?
- Any other improvements or changes you'd like to see?

I'm happy to make any adjustments or additions based on your review!
Thanks for taking the time to review this PR.

**Review by viceice — DISMISSED** [review]

File: lib/workers/repository/update/branch/index.spec.ts [hunk 1]
```diff
@@ -1103,6 +1103,88 @@ describe('workers/repository/update/branch/index', () => {
       commit.commitFilesToBranch.mockResolvedValueOnce(null);
       await branchWorker.processBranch(config);
       expect(platform.ensureComment).toHaveBeenCalledTimes(1);
```
Comment (RahulGautamSingh) [inline lib/workers/repository/update/branch/index.spec.ts]:
```suggestion
```
The new assertion covers this

File: lib/workers/repository/update/branch/index.spec.ts [hunk 2]
```diff
@@ -1103,6 +1103,88 @@ describe('workers/repository/update/branch/index', () => {
...
+      const customMessage =
+        'Please check the lock file changes carefully before merging.';
+      const inconfig = {
+        ...config,
+        userStrings: {
+          artifactErrorWarning: customMessage,
+        },
+      };
+      await branchWorker.processBranch(inconfig);
+      expect(platform.ensureComment).toHaveBeenCalledTimes(1);
```
Comment (RahulGautamSingh) [inline lib/workers/repository/update/branch/index.spec.ts]:
```suggestion
```

File: lib/workers/repository/update/branch/index.spec.ts [hunk 3]
```diff
@@ -1103,6 +1103,88 @@ describe('workers/repository/update/branch/index', () => {
...
+      commit.commitFilesToBranch.mockResolvedValueOnce(null);
+      const inconfig = {
+        ...config,
+        userStrings: {
+          artifactErrorWarning:
+            'Lock file error in {{manager}} - please review carefully.',
+        },
+      };
+      await branchWorker.processBranch(inconfig);
+      expect(platform.ensureComment).toHaveBeenCalledTimes(1);
```
Comment (RahulGautamSingh) [inline lib/workers/repository/update/branch/index.spec.ts]:
```suggestion
```

File: lib/workers/repository/update/branch/index.ts [hunk 4]
```diff
@@ -870,11 +870,13 @@ export async function processBranch(
           { artifactErrors: config.artifactErrors },
           'artifactErrors',
         );
+        // TODO #22198
+        const userStrings = config.userStrings!;
```
Comment (RahulGautamSingh) [inline lib/workers/repository/update/branch/index.ts]:
```suggestion
```

File: lib/workers/repository/update/branch/index.ts [hunk 5]
```diff
@@ -870,11 +870,13 @@ export async function processBranch(
...
         let content = `Renovate failed to update `;
         content +=
           config.artifactErrors.length > 1 ? 'artifacts' : 'an artifact';
-        content +=
-          ' related to this branch. You probably do not want to merge this PR as-is.';
+        content += ' related to this branch. ';
+        content += template.compile(userStrings.artifactErrorWarning, config);
```
Comment (RahulGautamSingh) [inline lib/workers/repository/update/branch/index.ts]:
```suggestion
        content += template.compile(config.userStrings!.artifactErrorWarning, config);
```

**Review by RahulGautamSingh — CHANGES_REQUESTED** [review]
Minor nitpicking. All else looks good.

**Comment by hesstobi** [comment]
@RahulGautamSingh Thank you for your review. I added your suggestions.

**Review by RahulGautamSingh — APPROVED** [review]

**Review by rarkins — APPROVED** [review]
